cupy/random/_generator_api.pyx

```diff
@@ -262,6 +265,54 @@ class Generator:
         """
         return self.standard_exponential(size) * scale
 
+    def geometric(self, p, size=None):
+        """Geometric distribution.
+
+        Returns an array of samples drawn from the geometric distribution. Its
+        probability mass function is defined as
+
+        .. math::
+            f(x) = p(1-p)^{k-1}.
+
+        Args:
+            p (float): Success probability of the geometric distribution.
```

Comment:
p can be an array of floats
check numpy

https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.geometric.html#numpy.random.Generator.geometric